### PR TITLE
Process recursive folder for spiffs images

### PR DESCRIPTION
**Pull request by tuanpmt**
Process recursive folder
```
+files
    +--folder1
    |   \---file1.txt
    |   \---file2.txt
    \--file3.txt
```
to spiffs filename:

```
+ folder1/file1.txt
+ folder1/file2.txt
+ file3.txt
```

### CODE REVIEW

File: extras/spiffs/mkspiffs/mkspiffs.c [hunk 1]
```diff
@@ -187,12 +187,27 @@ static bool process_directory(const char *direcotry)
...
-            printf("Processing file %s\n", path);
-            if (!process_file(path, ep->d_name)) {
+            filename = path;
+            file_ptr = path;
+            while(*file_ptr != 0 && *file_ptr != '/') {
+                file_ptr ++;
+            }
+            if(*file_ptr != 0) {
+                filename = file_ptr + 1;
+            }
```
Comment (sheinz) [inline extras/spiffs/mkspiffs/mkspiffs.c]: As far as I understand this block of code sets `filename` to point past the fist occurrence  of '/'.
I may suggest alternative implementation:
```c
filename = strchr(path, '/');
filename = filename ? &filename[1] : path;
```

Comment (tuanpmt) [inline extras/spiffs/mkspiffs/mkspiffs.c]: Ok, i did it
Thanks

File: extras/spiffs/mkspiffs/mkspiffs.c [hunk 2]
```diff
@@ -178,7 +178,7 @@ static bool process_directory(const char *direcotry)
 {
     DIR *dp;
     struct dirent *ep;
-    char path[256];
+    char path[256], *filename, *file_ptr;
```
Comment (sheinz) [inline extras/spiffs/mkspiffs/mkspiffs.c]: `file_ptr` is no longer needed.

File: extras/spiffs/mkspiffs/mkspiffs.c [hunk 3]
```diff
@@ -187,12 +187,23 @@ static bool process_directory(const char *direcotry)
...
+                continue;
+            }
             if (ep->d_type != DT_REG) {
                 continue;  // not a regular file
             }
             sprintf(path, "%s/%s", direcotry, ep->d_name);
-            printf("Processing file %s\n", path);
-            if (!process_file(path, ep->d_name)) {
+            filename = path;
+            filename = strchr(path, '/');
```
Comment (sheinz) [inline extras/spiffs/mkspiffs/mkspiffs.c]: `filename = path` may be removed as the next line will overwrite it anyway.

**Comment by tuanpmt** [comment]
@sheinz as you said

**Comment by sheinz** [comment]
Thanks @tuanpmt !

**Pull request merged**